src/material/snack-bar/snack-bar-ref.ts

```diff
@@ -17,6 +17,9 @@ export interface MatSnackBarDismiss {
   dismissedByAction: boolean;
 }
 
+/** Maximum amount of milliseconds that can be passed into setTimeout. */
+const MAX_TIMEOUT = Math.pow(2, 31) - 1;
```

Comment:
Why doesn't `Number.MAX_SAFE_INTEGER` work for this?

Reply:
AFAIK `MAX_SAFE_INTEGER` is 64bit, whereas `setTimeout` accepts 32bit.